sdk/monitor/Azure.ResourceManager.Monitor/api/Azure.ResourceManager.Monitor.netstandard2.0.cs

```diff
@@ -648,7 +654,7 @@ public partial class MonitorPrivateEndpointConnectionData : Azure.ResourceManage
         public MonitorPrivateEndpointConnectionData() { }
         public Azure.ResourceManager.Monitor.Models.MonitorPrivateLinkServiceConnectionState ConnectionState { get { throw null; } set { } }
         public Azure.Core.ResourceIdentifier PrivateEndpointId { get { throw null; } }
-        public Azure.ResourceManager.Monitor.Models.MonitorPrivateEndpointConnectionProvisioningState? ProvisioningState { get { throw null; } }
+        public Azure.ResourceManager.Monitor.Models.MonitorPrivateEndpointConnectionProvisioningState? ProvisioningState { get { throw null; } set { } }
```

Comment:
Is it correct that a privisioning state has a setter? Should this be readonly?

Reply:
No, ProvisioningState is defined as readonly in our Swagger. How can I update this in the autorest, assuming it is possible?

I thought it may have been an issue specific to https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v4/privatelinks.json, [which is ref'd by `privateEndpointConnections` in the Swagger](https://github.com/Azure/azure-rest-api-specs/blob/69df3c5d425f59599fa3cebd14e9843d5b82a91f/specification/monitor/resource-manager/Microsoft.Monitor/stable/2023-04-03/monitoringAccounts_API.json#LL456C12-L456C12). But even that appears to be readonly.

Reply:
Maybe this is a modelerfour bug, but writing readonly inside the model definition does not work.
The readonly true should be defined on this property: https://github.com/Azure/azure-rest-api-specs/blob/a2140e7ff0a6c28df501b16c51fe85b762a386fc/specification/common-types/resource-management/v4/privatelinks.json#L55
We should be able to add the readonly using directive.

Reply:
I'm still looking into it, but I've tried adding this directive and similar variations, without any luck so far
```
  - from: v4/privatelinks.json
    where: $.definitions.PrivateEndpointConnectionProvisioningState
    transform: $["readOnly"] = true
```

Reply:
Okay, after a few snags with successfully changing Line 657 but removing other necessary lines, we've arrived at the solution with
```
  - from: swagger-document
    where: $.definitions.PrivateEndpointConnectionProperties.properties.provisioningState
    transform: $["readOnly"] = true
```